### DISCUSSION TITLE
Understanding collaboration links

### DISCUSSION THREAD

**Discussion by inwardmovement** [excalidraw/excalidraw#2787] [Q&A] [answered]
Some things are not clear to me:

- Is the room link stored somewhere I can find it easily, after closing the collaboration session? or do I have to save it on my own if I want to access/share the room later?

- If every room users close their collaboration session, does the link persist?

- When does the room link change: each time I disable/enable collaboration? or each time I disable/enable collaboration after the canvas has changed?

- Couldn't we provide the user a UI to access their rooms easily?

I'm trying to figure out how exactly collaboration sessions/links work in Excalidraw, comparing to how it works in Google Documents for example.

**Comment by dwelle** [comment, marked as answer]
> Is the room link stored somewhere I can find it easily, after closing the collaboration session? or do I have to save it on my own if I want to access/share the room later?

Yes, you need to store the link yourself (e.g. you can bookmark it). The link contains the encryption `key` which you need to decrypt the scene. The key is private and not transmitted to our servers, so even if we decided to publish a list of rooms (not something we'll do or want to do), it wouldn't be of any use to you without the key.

> If every room users close their collaboration session, does the link persist?

Yes, the room & its contents is persisted to the server, in an encrypted format. You can go back to it anytime in the future.

> When does the room link change: each time I disable/enable collaboration? or each time I disable/enable collaboration after the canvas has changed?

A new room link is generated every time you start the session via the button in the room dialog.

> Couldn't we provide the user a way of accessing "his rooms" easily?

Technically yes. But it'd be stored in `localStorage`, so would not be very permanent, nor portable (across browsers, devices...). It'd also incentivize using collaboration rooms in ways they were not designed for at this point. We'll see in the future.

**Reply by inwardmovement** [reply]
Thanks for the replies!

**Reply by inwardmovement** [reply]
Can I make a PR to clarify the "Share a live-collaboration session" (session started) screen to reflect those elements?

**Reply by dwelle** [reply]
Hard to say. Current sentiment within the team is the dialog may already contain too much text, and we should reduce, not expand, so your PR will likely not get merged. You can try.

**Comment by inwardmovement** [comment]
@dwelle What do you think about this user flow:

**Proposed flow:**
- _Start live collaboration_
  - Save the room link so you can access it whenever you want.
  - The room link persists after the end of your collaborative session.
  - Restarting the collaborative session will generate a new link to a new room.
  - Your collaborative sessions are private thanks to end-to-end encryption.
  - _Start session_
    - Share this link with anyone you want to collaborate with: [link] [your-name]
    - _Stop session_

**Proposed flow (alternative):**
- _Start live collaboration_
  - _Start session_
  - (optional) How does it work? [collapse] (see below)
    - Share this link with anyone you want to collaborate with: [link] [your-name]
    - _Stop session_
    - How does it work? [collapse]
      - Save the room link so you can access it whenever you want.
      - The room link persists after the end of your collaborative session.
      - Restarting the collaborative session will generate a new link to a new room.
      - Your collaborative sessions are private thanks to end-to-end encryption.
 
**Proposed layout:**

[![image](https://user-images.githubusercontent.com/9438102/104618887-8cf25000-568d-11eb-8305-a870c7440dfb.png)](https://excalidraw.com/#json=5750420840382464,QIXLx8VXMDtJ8eRoiFIafA)

**Reply by dwelle** [reply]
I like the alternative version.

But maybe we should display at least something by default, as opposed to putting everything under the details. Also the `how does it work` details should probably contain a bit different info before and after you start a session, but we could work it out in the PR.

THAT SAID, we'd still need to discuss this in the team as a whole. Maybe create an issue proposal?
